tests/Notifiers/Slack/SlackNotifierTest.php

```diff
@@ -0,0 +1,168 @@
+
+        $testConference = $this->createDummyConference(new \DateTime(), false);
+
+        yield 'Test 1 new conferences and 0 ending Cfps' => [
+            'newConferences' => [$testConference],
+            'endingCfps' => [],
+            // One should say no conference to display, the other no CFP to display.
+            'expectedSections' => 2,
+        ];
+
+        yield 'Test 2 conferences and 2 ending Cfps' => [
+            'newConferences' => [$testConference, $testConference],
+            'endingCfps' => [$testConference, $testConference],
+            // There should be one for each new conference and CFP, and one for the CFP milestone.
+            'expectedSections' => 5,
+        ];
+
+        $cfp0days = $this->createDummyConference(new \DateTime(), false, false);
+        $cfp1days = $this->createDummyConference(new \DateTime('+1 day'), false, false);
+        $cfp5days = $this->createDummyConference(new \DateTime('+5 days'), false, false);
+        $cfp10days = $this->createDummyConference(new \DateTime('+10 days'), false, false);
+        $cfp20days = $this->createDummyConference(new \DateTime('+20 days'), false, false);
+        $cfp30days = $this->createDummyConference(new \DateTime('+30 days'), false, false);
+
+        yield 'Test all CFP end date milestones work' => [
+            'newConferences' => [],
+            'endingCfps' => [$cfp0days, $cfp1days, $cfp5days, $cfp10days, $cfp20days, $cfp30days],
+            // There should be one telling no conferences to display, and one for each CFP milestone + one for corresponding CFP.
+            'expectedSections' => 13,
+        ];
+
+        $excludedConference = $this->createDummyConference(new \DateTime(), true);
+
+        yield 'Test excluded conference is not sending notifications' => [
+            'newConferences' => [$testConference, $excludedConference],
+            'endingCfps' => [],
+            // One for $testConference, one to say no ending CFP.
+            'expectedSections' => 2,
+        ];
+    }
+
+    private function createSlackNotifier(array $endingCfps): SlackNotifier
+    {
+        $webHookUrl = 'Slack Webhook';
+        $httpClient = new MockHttpClient();
+
+        $router = $this->prophesize(Router::class);
+        $router
+            ->generate(
+                Argument::type('string'),
+                [
+                    'id' => 1,
+                    'entity' => 'NextConference',
+                    'action' => 'show',
+                ],
+                Argument::type('integer')
+            )
+            ->willReturn('Route URL');
+
+        $conferenceRepository = $this->prophesize(ConferenceRepository::class);
+        $conferenceRepository
+            ->findEndingCfps()
+            ->willReturn($endingCfps);
+
+        return new SlackNotifier(
+            $webHookUrl,
+            $httpClient,
+            $router->reveal(),
+            $conferenceRepository->reveal(),
+            new SlackBlocksBuilder(),
```

Comment:
Sometimes you create variable (and you don't alter them) and sometime not
`$webHookUrl` VS `new SlackBlocksBuilder()` for example.
This is not really consistant